app/controllers/api/v1/timelines/link_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Api::V1::Timelines::LinkController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :read, :'read:statuses' }, only: :show, if: :require_auth?
+
+  before_action :set_preview_card
+  before_action :set_statuses
+
+  after_action :insert_pagination_headers, unless: -> { @statuses.empty? }
+
+  def show
+    cache_if_unauthenticated!
+    render json: @statuses, each_serializer: REST::StatusSerializer, relationships: StatusRelationshipsPresenter.new(@statuses, current_user&.account_id)
+  end
+
+  private
+
+  def require_auth?
+    !Setting.timeline_preview
+  end
+
+  def set_preview_card
+    @preview_card = PreviewCard.joins(:trend).merge(PreviewCardTrend.allowed).find_by!(url: params[:url])
+  end
+
+  def set_statuses
+    @statuses = @preview_card.nil? ? [] : preload_collection(link_timeline_statuses, Status)
+  end
+
+  def link_timeline_statuses
+    link_feed.get(
+      limit_param(DEFAULT_STATUSES_LIMIT),
+      params[:max_id],
+      params[:since_id],
+      params[:min_id]
+    )
+  end
+
+  def link_feed
+    LinkFeed.new(
+      @preview_card,
+      current_account,
+      local: truthy_param?(:local),
+      remote: truthy_param?(:remote)
+    )
+  end
+
+  def insert_pagination_headers
+    set_pagination_headers(next_path, prev_path)
+  end
+
+  def pagination_params(core_params)
+    params.slice(:url, :local, :remote, :limit).permit(:url, :local, :remote, :limit).merge(core_params)
+  end
+
+  def next_path
+    api_v1_timelines_link_url pagination_params(max_id: pagination_max_id)
+  end
+
+  def prev_path
+    api_v1_timelines_link_url pagination_params(min_id: pagination_since_id)
+  end
+
+  def pagination_max_id
```

Comment:
For this and the `_since_id` method below, these are defined in `Api::Pagination` concern - you can define `pagination_collection` (would just be `@statuses` here) and the module will use that for max/since methods.